### PR TITLE
pyperf: Support Github Markdown table format

### PR DESCRIPTION
**Pull request by corona10**
closes: https://github.com/psf/pyperf/issues/113

### CODE REVIEW

**Comment by corona10** [comment]
cc @pablogsal @brettcannon

**Comment by brettcannon** [comment]
Do you happen to have an example table to paste into a comment to see what the output is like?

**Comment by corona10** [comment]
@brettcannon 
Sure :)

스크린샷 2021-09-26 오전 6 23 13

https://gist.github.com/corona10/6db6701efdc4769c237d3d3de5342489

File: doc/cli.rst [hunk 1]
```diff
@@ -101,12 +102,16 @@ Options:
...
 * ``--table``: Render a table.
+* ``--table-format``: Table rendering format.
 * ``--benchmark NAME`` only displays the benchmark called ``NAME``. The option
   can be specified multiple times.
 
 .. versionchanged:: 1.2
    The ``--benchmark`` option can now be specified multiple times.
 
+.. versionchnaged:: 2.3
+   The ``--table-format`` option now can designate format between reST and markdown.
```
Comment (corona10) [inline doc/cli.rst:113]: designate or specify or choose?

**Comment by brettcannon** [comment]
@corona10 looks great!

File: pyperf/tests/test_perf_cli.py [hunk 2]
```diff
@@ -154,6 +154,26 @@ def test_compare_to_table(self):
...
+
+        changed_result = self.create_bench((2.0,),
+                                           metadata={'name': 'telco'})
+
+        stdout = self.compare('compare_to', ref_result, changed_result, '--table',
+                              '--table-format', 'md')
+
+        expected = textwrap.dedent('''
+            | Benchmark | ref      | changed                |
+            |-----------|----------|------------------------|
```
Comment (vstinner) [inline pyperf/tests/test_perf_cli.py:170]: Would it be possible to align the first column to the left, and align other columns at the end?

`| --------- | :------: | :--------------------: |`

Comment (corona10) [reply pyperf/tests/test_perf_cli.py]: @vstinner Done :)

File: pyperf/tests/test_perf_cli.py [hunk 3]
```diff
@@ -154,6 +154,26 @@ def test_compare_to_table(self):
...
+
+        changed_result = self.create_bench((2.0,),
+                                           metadata={'name': 'telco'})
+
+        stdout = self.compare('compare_to', ref_result, changed_result, '--table',
+                              '--table-format', 'md')
+
+        expected = textwrap.dedent('''
+            | Benchmark | ref      | changed                |
+            |-----------|:--------:|:----------------------:|
```
Comment (vstinner) [inline pyperf/tests/test_perf_cli.py:170]: Does it work without spaces? In some MarkDown examples, I saw:

```suggestion
            | --------- | :------: | :--------------------: |
```

**Review by vstinner — APPROVED** [review]
LGTM. You may mention in the doc/changelog.rst.

File: pyperf/tests/test_perf_cli.py (hunk 3 above)
Comment (corona10) [reply pyperf/tests/test_perf_cli.py:170]: > Does it work without spaces? In some MarkDown examples, I saw:

yes it works.
see https://gist.github.com/corona10/6db6701efdc4769c237d3d3de5342489

Comment (vstinner) [reply pyperf/tests/test_perf_cli.py:170]: I like centered values, it's nice :-)

**Pull request merged**